src/sentry/static/sentry/app/components/issueDiff/index.tsx

```diff
@@ -0,0 +1,190 @@
+import getStacktraceBody from 'app/utils/getStacktraceBody';
+import SplitDiff from 'app/components/splitDiff';
+import ButtonBar from 'app/components/buttonBar';
+import Button from 'app/components/button';
+import space from 'app/styles/space';
+import {Project} from 'app/types';
+
+import renderGroupingInfo from './groupingDiff';
+
+type Props = {
+  api: Client;
+  baseIssueId: string;
+  targetIssueId: string;
+
+  orgId: string;
+  project: Project;
+
+  baseEventId?: string;
+  targetEventId?: string;
+  className?: string;
+};
+
+type State = {
+  loading: boolean;
+  groupingDiff: boolean;
+  baseEvent: Array<string>;
+  targetEvent: Array<string>;
+  SplitDiffAsync?: typeof SplitDiff;
+};
+
+class IssueDiff extends React.Component<Props, State> {
+  static defaultProps = {
+    baseEventId: 'latest',
+    targetEventId: 'latest',
+  };
+
+  constructor(props: Props) {
+    super(props);
+
+    this.state = {
+      loading: true,
+      groupingDiff: false,
+      baseEvent: [],
+      targetEvent: [],
+
+      // `SplitDiffAsync` is an async-loaded component
+      // This will eventually contain a reference to the exported component from `./splitDiff`
+      SplitDiffAsync: undefined,
+    };
+  }
+
+  componentDidMount() {
+    this.fetchData();
+  }
+
+  fetchData() {
+    const {baseIssueId, targetIssueId, baseEventId, targetEventId} = this.props;
+
+    // Fetch component and event data
+    Promise.all([
+      import(/* webpackChunkName: "splitDiff" */ '../splitDiff'),
+      this.fetchEventData(baseIssueId, baseEventId || 'latest'),
+      this.fetchEventData(targetIssueId, targetEventId || 'latest'),
+    ])
+      .then(([{default: SplitDiffAsync}, baseEvent, targetEvent]) => {
+        this.setState({
+          SplitDiffAsync,
+          baseEvent,
+          targetEvent,
+          loading: false,
+        });
+      })
+      .catch(() => {
+        addErrorMessage(t('Error loading events'));
+      });
+  }
+
+  toggleDiffMode = () => {
+    this.setState(
+      state => ({groupingDiff: !state.groupingDiff, loading: true}),
+      this.fetchData
+    );
+  };
+
+  fetchEventData = async (issueId: string, eventId: string) => {
+    const {orgId, project, api} = this.props;
+    const {groupingDiff} = this.state;
+
+    if (eventId === 'latest') {
+      const event = await api.requestPromise(`/issues/${issueId}/events/latest/`);
+      eventId = event.eventID;
+    }
+
+    if (groupingDiff) {
+      const groupingInfo = await api.requestPromise(
+        `/projects/${orgId}/${project.slug}/events/${eventId}/grouping-info/`
+      );
+      return renderGroupingInfo(groupingInfo);
+    } else {
+      const event = await api.requestPromise(
+        `/projects/${orgId}/${project.slug}/events/${eventId}/`
+      );
+      return getStacktraceBody(event);
+    }
+  };
+
+  render() {
+    const {className, project} = this.props;
+    const {
+      SplitDiffAsync: DiffComponent,
```

Comment:
I think the name `SplitDiffAsync`. It is also fine if you only call it `SplitDiff`

Reply:
it conflicts with an import of the same name